### ISSUE TITLE
vpg: possible redundant code

### ISSUE THREAD

**Issue by scotthuang1989** [openai/spinningup#64]
in vpg.py.

the function: `mlp_actor_critic `

will receive observations and ation

> x, **a**

return 4 items:

> pi, logp, logp_pi, v
pi: action choose by policy
logp_pi: log probability of pi
logp: **log probability of input action a**
v: value of this x according to value network.

when running the trajectory, vpg will evaluate pi and logp_pi of the compuation graph, and save the returned action and it's log probability.
pi is the action we will take in environment. logp_pi is its log probability.

when update the policy,  vpg will evaluate following item:

`pi_loss = -tf.reduce_mean(logp * adv_ph)`

this is said, vpg will feed action to computation graph and get it's log probability, but since policy will only be updated once, so the log probability of action is the same as we running trajectory, we can just use our saved log probability.

Can any one clarity it if I am wrong?

**Comment by scotthuang1989** [comment]
sorry for the silly question. we need take gradient with respect to the parameter of policy network. so can't just  substitute a constant.
